com.unity.ml-agents.extensions/Runtime/Sensors/GridSensor.cs

```diff
@@ -536,73 +553,112 @@ protected void ChannelsToTexture(int channelIndex, int numChannelsToAdd)
         /// <returns>A float[] containing all of the information collected from the gridsensor</returns>
         public float[] Perceive()
         {
+            if (m_CollidersPerCell == null)
+            {
+                return Array.Empty<float>();
+            }
             Reset();
```

Comment:
I missed this before. ISensor.Reset() is meant to be called by the Agent when it resets, so that any stateful information doesn't get carried over when it dies (e.g. StackingSensor). We should probably rename the current GridSensor.Reset() method to something else (`ClearPerceptionBuffer()?`) and make Reset() empty.